### PR TITLE
added catch file construction as apart of generate_assets

### PR DESCRIPTION
**Pull request by DannyPeck**
- Hatch will now download Catch.hpp from the supplied link on Catch's readme: https://github.com/catchorg/Catch2/releases/download/v2.1.1/catch.hpp.
- Hatch now adds the catch.hpp and catch.cpp files to the `/project/test/src` directory.

### CODE REVIEW

File: hatch/src/assets/builder.rs [hunk 1]
```diff
@@ -119,4 +135,28 @@ impl Builder {
 
     ProjectAsset::new(project_path.to_path_buf(), Windows::name(), contents)
   }
+
+  pub fn catch_header(&self, project: &Project) -> HatchResult<ProjectAsset> {
+    let test_src_path = project.path().join("test/src");
+    let file_name = CatchHeader::name();
+    if !test_src_path.join(file_name).exists() {
+      let catch_header_url = "https://github.com/catchorg/Catch2/releases/download/v2.1.1/catch.hpp";
```
Comment (xmclark) [inline hatch/src/assets/builder.rs]: Could we put this special url string in a static location outside the function?

File: hatch/src/assets/builder.rs [hunk 2]
```diff
@@ -119,4 +135,28 @@ impl Builder {
...
+      let mut resp = reqwest::get(catch_header_url).with_context(|e| {
+        format!("failed to retrieve catch.hpp : {}", e)
+      })?;
+      let mut content = String::new();
+      resp.read_to_string(&mut content);
```
Comment (xmclark) [inline hatch/src/assets/builder.rs]: I think there is a .text() method on the Response object. You may be able to shorten this to
```
let resp = reqwest::get(catch_header_url).with_context(|e| {
  format!("failed to retrieve catch.hpp : {}", e)
})?;
let content = resp.text()?;
```

**Review by xmclark — DISMISSED** [review]

File: hatch/src/assets/builder.rs (hunk 1 above)
Comment (technetos) [reply hatch/src/assets/builder.rs]: I agree please make this static

**Review by technetos — DISMISSED** [review]

**Review by technetos — APPROVED** [review]

File: hatch/src/assets/builder.rs [hunk 3]
```diff
@@ -140,14 +141,16 @@ impl Builder {
     let test_src_path = project.path().join("test/src");
     let file_name = CatchHeader::name();
     if !test_src_path.join(file_name).exists() {
-      let catch_header_url = "https://github.com/catchorg/Catch2/releases/download/v2.1.1/catch.hpp";
-      let mut resp = reqwest::get(catch_header_url).with_context(|e| {
-        format!("failed to retrieve catch.hpp : {}", e)
+      let res = (|| -> HatchResult<ProjectAsset> {
+        let mut resp = reqwest::get(CATCH_HEADER_URL)?;
```
Comment (xmclark) [inline hatch/src/assets/builder.rs]: does this need to be `mut`?

**Review by xmclark — APPROVED** [review]
